app/src/main/java/de/danoeh/antennapod/fragment/AudioPlayerFragment.java

```diff
@@ -178,6 +180,30 @@ public void setHasChapters(boolean hasChapters) {
         this.hasChapters = hasChapters;
         tabLayoutMediator.detach();
         tabLayoutMediator.attach();
+        if (controller != null) {
+            setChapterDividers(controller.getMedia());
+        }
+    }
+
+    public void setChapterDividers(Playable media) {
+
+        if (media == null) {
+            return;
+        }
+
+        List<Chapter> chapters;
+        float[] dividerPos = null;
+
+        if ((chapters = media.getChapters()) != null && chapters.size() > 0) {
```

Comment:
I think it would be more clean to call `media.loadChapterMarks(getContext());` when loading the media item and then updating the bar in `updateUi`. This also ensures that `controller.getMedia` is always called in a background thread (it performs database access in some cases).

That would also allow for making the `setHasChapters` method private and can therefore reduce coupling between chapter fragment and player fragment.

Reply:
Fixed in the last commit